eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -48,10 +65,10 @@ steps:
 
               if ($repository.Commitish -match '^refs/pull/\d+/merge$') {
                 Write-Host "git clone --no-checkout --filter=tree:0 -c remote.origin.fetch='+$($repository.Commitish):refs/remotes/origin/$($repository.Commitish)' https://github.com/$($repository.Name) ."
```

Comment:
Should we move the command logging into this retry function now as well? Just a way to save some lines of our inline script and also reduce some duplication of the command.

Reply:
Yeah I could move all logging into the function, and make it a generic runner with a retries int, that way we can pass all commands through it.

Reply:
@weshaggard ok made this update here. Overdue anyway.